### DISCUSSION TITLE
Get IP address when user is logging in

### DISCUSSION THREAD

**Discussion by MasterWolfx** [nextauthjs/next-auth#8952] [Help]
### Summary

I need to get the ip address of the user who is loggin in.

### Additional information

At the moment i can get userAgent trough headers, i also find some headers that should contain the ip like x-forwarded-by or x-real-ip, but both of them are not set, and they can be easly spoofed. 
Has anynone any idea to get the ip address?

**Comment by codewithkashi** [comment]
You can get the IP address of the user in your Next.js app by following these steps:

### Step 1: Create an API Route to Get IP Address

**`pages/api/internal/getip.ts`**:

```
import { NextApiRequest, NextApiResponse } from 'next';
export default function handler(req: NextApiRequest, res: NextApiResponse) {
const ip = req.headers['x-forwarded-for'] || req.headers['x-real-ip'] || req.connection.remoteAddress || req.socket.remoteAddress;
res.status(200).json({ ip });
}
```

### Step 2: Fetch IP Address in the Login Component and Pass it to `signIn`

**`pages/login.tsx`**:

```
const response = await axios.post('/api/internal/getip');
const ip = response.data;
signIn('credentials', { username, password, ip });
```

### Step 3: Update NextAuth Configuration to Use the IP Address

**`pages/api/auth/[...nextauth].ts`**:

```
let { username, password, ip } = req.body;
```

### Explanation

1.  **API Route**: Captures and returns the client's IP address.
2.  **Login Component**: Fetches the IP address from the API and includes it in the sign-in request.
3.  **NextAuth Configuration**: Extracts the IP address from the credentials.
